### PR TITLE
feat: use event name instead of const to be psr-14 compliant

### PR DESCRIPTION
**Pull request by GautierLB**
## Why?
With symfony 4.3 and the psr-14, events are not supposed to rely on const for their types anymore, and should be indepent classes.

## How?
- Refactor events : 
   - Move existing code from `GuzzleCacheEvent` && `GuzzleHttpEvent` to `AbstractGuzzleCacheEvent` && `AbstractGuzzleHttpEvent` to stop relying on const for the event names (deprecated)
   - Create named events : 
      - `GuzzleHttpErrorEvent`
      - `GuzzleCacheErrorEvent`
- Update tests

### CODE REVIEW

File: src/EventDispatcher/AbstractGuzzleHttpEvent.php [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    protected $clientId;
+
+    /**
+     * Set request
+     *
+     * @param Request $request
+     *
+     * @return static
+     */
+    public function setRequest(RequestInterface $request)
```
Comment (b-viguier) [inline src/EventDispatcher/AbstractGuzzleHttpEvent.php]: Is it an inherited method? Else what about return type?

```suggestion
    public function setRequest(RequestInterface $request): self
```

File: src/EventDispatcher/AbstractGuzzleCacheEvent.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        $this->exception = $e;
+    }
+
+    /**
+     * Get Exception if isset
+     *
+     * @return mixed
+     */
+    public function getException()
```
Comment (b-viguier) [inline src/EventDispatcher/AbstractGuzzleCacheEvent.php]:
```suggestion
    public function getException(): ?\Exception
```

File: src/EventDispatcher/AbstractGuzzleHttpEvent.php [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+        return $this;
+    }
+
+    /**
+     * Return request
+     *
+     * @return Request
+     */
+    public function getRequest()
```
Comment (b-viguier) [inline src/EventDispatcher/AbstractGuzzleHttpEvent.php]:
```suggestion
    public function getRequest(): Request
```

File: src/EventDispatcher/AbstractGuzzleHttpEvent.php [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    /**
+     * Set Response
+     *
+     * @param Response $response
+     *
+     * @return static
+     */
+    public function setResponse(ResponseInterface $response)
```
Comment (b-viguier) [inline src/EventDispatcher/AbstractGuzzleHttpEvent.php]:
```suggestion
    public function setResponse(ResponseInterface $response): self
```

File: src/EventDispatcher/AbstractGuzzleHttpEvent.php [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+        return $this;
+    }
+
+    /**
+     * Return response
+     *
+     * @return Response
+     */
+    public function getResponse()
```
Comment (b-viguier) [inline src/EventDispatcher/AbstractGuzzleHttpEvent.php]:
```suggestion
    public function getResponse(): Response
```

File: src/EventDispatcher/AbstractGuzzleHttpEvent.php [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    {
+        return $this->request;
+    }
+
+    /**
+     * Set Response
+     *
+     * @param Response $response
+     *
+     * @return static
```
Comment (b-viguier) [inline src/EventDispatcher/AbstractGuzzleHttpEvent.php]: Why `static`?  (Same question for other usage in this PR)
Do you really need the cost of [LSB](https://www.php.net/manual/en/language.oop5.late-static-bindings.php)? Why `self` is not enough?

File: src/EventDispatcher/AbstractGuzzleHttpEvent.php [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+    }
+
+    /**
+     * Set reason
+     *
+     * @param mixed $reason
+     *
+     * @return static
+     */
+    public function setReason($reason)
```
Comment (b-viguier) [inline src/EventDispatcher/AbstractGuzzleHttpEvent.php]:
```suggestion
    public function setReason($reason): self
```

File: src/EventDispatcher/GuzzleHttpErrorEvent.php [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace M6Web\Bundle\GuzzleHttpBundle\EventDispatcher;
+
+/**
+ * Class GuzzleHttpErrorEvent
+ */
```
Comment (saiht) [inline src/EventDispatcher/GuzzleHttpErrorEvent.php]: useless class comment

File: src/EventDispatcher/AbstractGuzzleCacheEvent.php [hunk 9]
```diff
@@ -0,0 +1,72 @@
+<?php
+namespace M6Web\Bundle\GuzzleHttpBundle\EventDispatcher;
+
+use Psr\Http\Message\RequestInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * Class AbstractGuzzleCacheEvent
+ */
```
Comment (saiht) [inline src/EventDispatcher/AbstractGuzzleCacheEvent.php]: useless class comment

Comment (GautierLB) [reply src/EventDispatcher/AbstractGuzzleCacheEvent.php]: Sadly we're using coke as a cs-fixer on this project 🙁

**Comment by GautierLB** [comment]
There shouldn't be any BC break with previous symfony versions, we kept the compatibility so the `composer.json` doesn't need to be changed 
(for information, current symfony version in `composer.json` is : `"~2.3||~3.0||~4.0"`)

**Pull request merged**